pandas/core/generic.py

```diff
@@ -4545,17 +4545,16 @@ def pipe(self, func, *args, **kwargs):
 
     Parameters
     ----------
-    func : function, string, dictionary, or list of string/functions
+    func : function, string, list of string/functions or dictionary
```

Comment:
We try to use only Python types in this row, and be consistent with the format. `func : function, str, list or dict` would be the preferred format (or specifying the list types, like `list of str, list of function`).

Reply:
Alright, but the list can contain bot functions and strings (``[np.exp, 'sqrt']``). Maybe ``func : function, string, list of functions and/or strings or dict``?

Reply:
At some point I'd like to validate that all the types provided in that line are from a subset, to avoid typos and inconsistencies. I understand your point about mixing both, but I'd prefer `function, str, list or dict` (note `str`, the Python type, instead of `string`), and then provide the details about mixing strings and everything else in the description.